### PR TITLE
Fix #127: suggested time is no longer in the past

### LINKED ISSUE #127: Assign feature: suggested time is in the past
Screenshot 2019-10-30 at 12 11 12 PM

### PR DESCRIPTION
**Pull request by tysg**
Closes #127 
* Add a `GlobalClock` object: a static method can set the clock to always return a static time
* Add an entry in `config.json`: when `testEnv: true`, the clock will be set to the above static clock, and the program will run as if it's always a that fixed time
* Also add unit test for assign command

### CODE REVIEW

File: src/main/java/seedu/address/commons/core/Config.java [hunk 1]
```diff
@@ -32,6 +33,14 @@ public void setUserPrefsFilePath(Path userPrefsFilePath) {
...
     }
 
+    public boolean getIsTestEnv() {
+        return testEnv;
+    }
+
+    public void setTestEnv (boolean testEnv) {
+        this.testEnv = testEnv;
+    }
+
```
Comment (junnbang) [inline src/main/java/seedu/address/commons/core/Config.java]: is not being used through the codes, can refer to setuserPrefsFilePath and see how it is being used.
Maybe can insert a comment saying what it is for or how to use it

File: src/main/java/seedu/address/logic/commands/EditTaskCommand.java [hunk 2]
```diff
@@ -88,7 +89,8 @@ public CommandResult execute(Model model) throws CommandException {
         Task editedTask = createEditedTask(taskToEdit, editTaskDescriptor, model);
 
         //if the updated date is not same as original and if date is not today onwards
-        if (!taskToEdit.getDate().equals(editedTask.getDate()) && editedTask.getDate().isBefore(LocalDate.now(clock))) {
+        if (!taskToEdit.getDate().equals(editedTask.getDate())
+                && editedTask.getDate().isBefore(GlobalClock.dateToday())) {
```
Comment (junnbang) [inline src/main/java/seedu/address/logic/commands/EditTaskCommand.java]: Pls also remove the clock variables that it is currently using and remove the overload the constructor created for the clock

**Review by junnbang — CHANGES_REQUESTED** [review]
looks good, just minor adjustment

File: src/main/java/seedu/address/commons/core/Config.java (hunk 1 above)
Comment (tysng) [reply src/main/java/seedu/address/commons/core/Config.java]: Deleted the method

File: src/main/java/seedu/address/logic/commands/EditTaskCommand.java (hunk 2 above)
Comment (tysng) [reply src/main/java/seedu/address/logic/commands/EditTaskCommand.java]: Fixed

**Review by junnbang — APPROVED** [review]
Looks good!

**Pull request merged**